mixer/pkg/cache/cache.go

```diff
@@ -80,4 +79,7 @@ type ExpiringCache interface {
 	// The entry will be automatically expunged from the cache at or slightly after the
 	// requested expiration time.
 	SetWithExpiration(key interface{}, value interface{}, expiration time.Duration)
+
+	// EvictExpired() synchronously evicts all expired entries from the cache
+	EvictExpired()
```

Comment:
Since eviction is a delayed process, EvictExpired allows the user of the cache to forcibly remove any expired entries from the cache and thus avoiding serving stale stuff. It's probably not that useful in the common case, but it can be handy during debugging/diagnostics.

Remove is useful if you decide something is no longer true and shouldn't be in the cache anymore. 

RemoveAll is useful to reset everything without creating garbage. An LRU cache can be fairly large and it's not ideal to just throw it out and recreate it.

Reply:
I think Remove is required for many scenarios. RemoveAll is only about starting/stopping a cache without creating garbage. EvictExpired is indeed not that useful, except in tests. It's the same reason why there are methods in the runtime systems for Go, Java, C# that allow the programmer to explicitly trigger garbage collection. Sometimes, the user just knows better.